src/Readers/ODataReader.v4/OdcmReader.cs

```diff
@@ -117,6 +118,16 @@ private void AddVocabularyAnnotations(OdcmObject odcmObject, IEdmVocabularyAnnot
                 odcmObject.Annotations = ODataVocabularyReader.GetOdcmAnnotations(_edmModel, annotatableEdmEntity).ToList();
                 odcmObject.Description = _edmModel.GetDescriptionAnnotation(annotatableEdmEntity);
                 odcmObject.LongDescription = _edmModel.GetLongDescriptionAnnotation(annotatableEdmEntity);
+
+                if(annotatableEdmEntity is IEdmEntitySet && odcmObject is OdcmProperty)
+                {
+                    OdcmCapability.SetCapabilitiesForEntitySet((OdcmProperty)odcmObject);

```

Comment:
At the time of construction of an OdcmProperty, we will not have enough knowledge of the restriction annotations on it. An example being a navigation property whose restriction annotation is not defined on it but rather on an entity set (via NonInsertableNavigationProperties etc,). 

Also reading of raw annotations is something which is not done in the Vipr.Core library. It is left to the OdataReader to read them and convert into OdcmAnnotations